Unify test framework so all use XUnit

devlead opened the pull request:
Currently different test frameworks are used for the test projects, I propose all should use XUnit.

msftclas commented:
Hi __@devlead__, I'm your friendly neighborhood Microsoft Pull Request Bot (You can call me MSBOT). Thanks for your contribution!
    <span>
        In order for us to evaluate and accept your PR, we ask that you sign a contribution license agreement. It's all electronic and will take just minutes. I promise there's no faxing. https://cla.microsoft.com.
    </span>

TTYL, MSBOT;

msftclas commented:
__@devlead__, Thanks for signing the contribution license agreement so quickly! Actual humans will now validate the agreement and then evaluate the PR.
<br />Thanks, MSBOT;

jasonwilliams200OK commented on tests/Microsoft.Fx.Portability.Offline.Tests/OffLineApiPortServiceTests.cs:
```diff
@@ -29,23 +28,25 @@ public class OffLineApiPortServiceTests
 
             var result = _OfflineApiPortService.QueryDocIdsAsync(expectedDocIds).Result.Response;
-            Assert.AreEqual(0, result.Count());
+            Assert.Equal(0, result.Count());
```
`Assert.Empty`

devlead replied:
Good call, corrected. Wanted to change as little as possible.

conniey commented on tests/Microsoft.Fx.Portability.Offline.Tests/OffLineApiPortServiceTests.cs:
```diff
@@ -29,23 +28,25 @@ public class OffLineApiPortServiceTests
 
-        [TestMethod]
-        [ExpectedException(typeof(ArgumentNullException))]
+        [Theory]
```
This should be `[Fact]` because `[Theory]` expects some parameterized input. As a result, the tests fail for me, as is but work when I change them to Fact.

devlead replied:
@conniey correct, I've corrected that error now.

twsouthwick commented on tests/Microsoft.Fx.Portability.Offline.Tests/OffLineApiPortServiceTests.cs:
```diff
@@ -29,23 +28,25 @@ public class OffLineApiPortServiceTests
         public void QueryDocIdsWithNullThrowsArgumentNullException()
         {
-            var result = _OfflineApiPortService.QueryDocIdsAsync(null).Result;
+            var result = Record.Exception(
```
Why is Assert.Throws<ArgumentNullException> not used?

devlead replied:
Corrected.

twsouthwick commented on tests/Microsoft.Fx.Portability.Offline.Tests/OffLineApiPortServiceTests.cs:
```diff
@@ -29,23 +28,25 @@ public class OffLineApiPortServiceTests
         {
             var expectedDocIds = new List<string>();
 
             var result = _OfflineApiPortService.QueryDocIdsAsync(expectedDocIds).Result.Response;
```
Xunit supports async tests, so removing the .Result with await would be awesome as well.

devlead replied:
I've changed this to use async.

jasonwilliams200OK replied:
@devlead, is your commit overwritten or something? AFAICT, it is still synchronous. :smile_cat:

jasonwilliams200OK replied:
Note that there are more than one occurrences of `.Result` in this delta.

twsouthwick replied:
Yes, I was referring to all the occurrences - I just comment on one :)

devlead replied:
I've now changed all ;)

jasonwilliams200OK replied:
:+1: :tada:

devlead commented:
@conniey @twsouthwick @jasonwilliams200OK Think, I've now addressed all issues and rebased, so should be good to go. Let me know if you have anymore feedback.

twsouthwick commented:
Thanks, @devlead for this unification! LGTM

conniey commented:
LGTM! :+1:
